cadcutils/cadcutils/net/groups_client.py

```diff
@@ -0,0 +1,587 @@
+#         x509 cert and available as a default for user_id in other
+#         method calls.
+#         certfile -- Path to CADC proxy certificate
+#         """
+#
+#         # This client does not support name/password authentication
+#         super(UsersClient, self).__init__(usenetrc=False, *args, **kwargs)
+#
+#         # Specific base_url for AC webservice
+#         host = os.getenv('AC_WEBSERVICE_HOST', self.host)
+#         path = os.getenv('AC_WEBSERVICE_PATH', '/ac')
+#         self.base_url = '%s://%s%s' % ('https', host, path)
+#         self.logger.info('Base URL ' + self.base_url)
+#
+#         # This client will need the user DN
+#         self.current_user_dn = self.get_current_user_dn()
+#
+#         # Specialized exceptions handled by this client
+#         self._HTTP_STATUS_CODE_EXCEPTIONS[404] = {
+#             "User": exceptions.UserNotFoundException(),
+#             "Group": exceptions.GroupNotFoundException()
+#         }
+#         self._HTTP_STATUS_CODE_EXCEPTIONS[409] = \
+#             exceptions.GroupExistsException()
+#
+#     def get_user(self, identity):
+#
+#         if identity is None:
+#             raise ValueError("User Identity cannot be None.")
+#         if not isinstance(identity, Identity):
+#             raise ValueError("identity must be of type Identity.")
+#
+#         url = "{0}/users/{1}?idType={2}".format(self.base_url,
+#         identity.name, identity.type)
+#         xml_string = self._download_xml(url)
+#         reader = UserReader()
+#         user = reader.read(xml_string)
+#         self.logger.info('Retrieved user {0}'.format(user.internal_id))
+#         return user
+
+
+def print_group(group):
+    print('Group:')
+    print('             ID: {}'.format(group.group_id))
+    print('    Description: {}'.format(group.description))
+    print('          Owner: {}'.format(group.owner.identities['X500'].name))
+    print('  Last Modified: {}'.format(group.last_modified))
+    print('    User Admins: {}'.format((',\n'+' '*17).join(
+        ['{} ({})'.format(
+            a.identities['HTTP'].name,
+            a.identities['X500'].name) for a in group.user_admins])))
+    print('   Group Admins: {}'.format(
+        (',\n'+' '*17).join([gr.group_id for gr in group.group_admins])))
+    print_group_members(group)
+
+
+def print_group_members(group):
+    print('   User Members: {}'.format((',\n'+' '*17).join(
+        ['{} ({})'.format(
+            m.identities['HTTP'].name,
+            m.identities['X500'].name) for m in group.user_members])))
+    print('  Group Members: {}'.format(
+        (',\n'+' '*17).join([gr.group_id for gr in group.group_members])))
+
+
+def main_app():
+    parser = util.get_base_parser(version=version.version, auth_required=True,
+                                  default_resource_id=CADC_AC_SERVICE_ID)
+
+    parser.description = (
+        'Client for access the User Group Management System (GMS) at the '
```

Comment:
"for access the" => "to access the" or "for accessing the"?

Doing grammar in this review because it's a first release.

Reply:
thank you